From f36f4f4105a4bf86664a9e7404751ee604f93c2f Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Wed, 3 Dec 2014 07:51:12 +0100
Subject: [PATCH] opt: net: wifi: allow building without wi-fi HAL

 * For example devices with older broadcom chips with driver built
   as module doesn't like this. When the device goes into suspend
   mode, the wifi service crashes and the device soft reboots.

 * To not build the HAL, use:
   BOARD_NO_WIFI_HAL := true

Change-Id: Ia186ce3a59a082cfe98d283492fa450014af52f4

Conflicts:
	service/Android.mk
---
 service/Android.mk | 23 ++++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/service/Android.mk b/service/Android.mk
index 08b9ff0..76b47a4 100755
--- a/service/Android.mk
+++ b/service/Android.mk
@@ -42,17 +42,18 @@ include $(BUILD_STATIC_LIBRARY)
 # ============================================================
 LIB_WIFI_HAL := libwifi-hal
 
-ifeq ($(BOARD_WLAN_DEVICE), bcmdhd)
-  LIB_WIFI_HAL := libwifi-hal-bcm
-else ifeq ($(BOARD_WLAN_DEVICE), qcwcn)
-  LIB_WIFI_HAL := libwifi-hal-qcom
-else ifeq ($(BOARD_WLAN_DEVICE), mrvl)
-  # this is commented because none of the nexus devices
-  # that sport Marvell's wifi have support for HAL
-  # LIB_WIFI_HAL := libwifi-hal-mrvl
-else ifeq ($(BOARD_WLAN_DEVICE), MediaTek)
-  # support MTK WIFI HAL
-  LIB_WIFI_HAL := libwifi-hal-mt66xx
+ifneq ($(BOARD_NO_WIFI_HAL), true)
+  ifeq ($(BOARD_WLAN_DEVICE), bcmdhd)
+    LIB_WIFI_HAL := libwifi-hal-bcm
+  else ifeq ($(BOARD_WLAN_DEVICE), qcwcn)
+    # this is commented because none of the nexus devices
+    # that sport Qualcomm's wifi have support for HAL
+    # LIB_WIFI_HAL := libwifi-hal-qcom
+  else ifeq ($(BOARD_WLAN_DEVICE), mrvl)
+    # this is commented because none of the nexus devices
+    # that sport Marvell's wifi have support for HAL
+    # LIB_WIFI_HAL := libwifi-hal-mrvl
+  endif
 endif
 
 # Build the HalUtil
-- 
1.9.1

